src/ts/types/networks/cardano.d.ts

```diff
@@ -55,45 +56,113 @@ export interface CardanoAddress {
 
 // Sign transaction
 
-
 export interface CardanoInput {
     path: string | number[];
     prev_hash: string;
     prev_index: number;
 }
+
+export type CardanoToken = {
+    assetNameBytes: string;
+    amount: string;
+}
+
+export type CardanoAssetGroup = {
+    policyId: string;
+    tokenAmounts: CardanoToken[];
+}
+
 export type CardanoOutput =
     | {
           addressParameters: CardanoAddressParameters;
           amount: string;
+          tokenBundle?: CardanoAssetGroup[];
       }
     | {
           address: string;
           amount: string;
-      };
+          tokenBundle?: CardanoAssetGroup[];
+      }
+
+export type CardanoPoolOwner = {
+    stakingKeyPath?: string | number[];
+    stakingKeyHash?: string;
+}
+
+export type CardanoPoolRelay = {
+    type: CardanoPoolRelayType;
+    ipv4Address?: string;
+    ipv6Address?: string;
+    port?: number;
+    hostName?: string;
+}
+
+export type CardanoPoolMetadata = {
+    url: string;
+    hash: string;
+}
+
+export type CardanoPoolMargin = {
+    numerator: string;
+    denominator: string;
+}
+
+export type CardanoPoolParameters = {
```

Comment:
I'm confused. Why does this PR touch pool registration at all? Catalyst has nothing to do with it.

Reply:
Because the types were missing from the typescript definitions so I added them. Maybe it should be at least in a separate commit.